doc/qbk/03_core/1_refresher.qbk

```diff
@@ -0,0 +1,200 @@
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+    Official repository: https://github.com/boostorg/beast
+]
+
+[section:asio_refresher Refresher]
+
+To use Beast effectively, a prior understanding of networking is required.
+This section reviews networking concepts as a reminder and guide for further
+learning.
+
+A
+['network]
+allows programs located anywhere to exchange information after opting-in
+to communications by establishing a
+['connection].
+Data is reliably transferred on a connection in either direction (full-duplex)
+with bytes arriving in the same order they were sent. These connections, along
+with the objects and types used to represent them, are collectively termed
+['streams].
+
+The
+['internet]
+is a global network of interconnected computers which exchange information
+using a variety of standardized communication protocols. The most popular
+protocol is
+['TCP/IP],
+which this library relies on exclusively. The protocol takes care of the
+low level details so that applications see a
+['stream],
+which is the reliable, full-duplex connection carrying the ordered set
+of bytes described above.
+
+A vendor supplies a program called a
+['driver],
+enabling networking hardware such as an
+['ethernet adaptor]
+to talk to the operating system. The OS in turn permits running programs to
+interact with networking using various flavors of interfaces, such as:
+
+* Berkeley sockets
+* POSIX sockets
+* Windows Sockets 2 ("Winsock")
+
+C++ Networking, represented by __NetTS__ and __Asio__, provides another layer
+of abstraction with features such as:
+
+* Deadline timers
+* Buffer sequences
+* Stream concepts
+* Asynchronous I/O
+
+These concepts enable generic programming so that higher levels of abstraction
+may be composed to arbitrary degree. In fact, the interfaces and concepts
+offered by networking are best described as providing support for general
+input and output algorithms ("I/O"), including TCP/IP.
+
+[heading Buffers]
+
+A
+['buffer]
+holds a contiguous sequence of bytes used to perform reads or writes on an
+['I/O object].
+The networking types __const_buffer__ and __mutable_buffer__ represent
+these memory regions as type-safe pointer/size pairs. The concepts
+__ConstBufferSequence__ and __MutableBufferSequence__ are bidirectional
+ranges whose value type is convertible to __const_buffer__ and
+__mutable_buffer__ respectively. Buffer sequences may be used to transact
+in multiple buffers in a single function call. These types are non-owning;
+copying a buffer or buffer sequences only creates a shallow reference,
+it does allocate a copy of the contents.
+
+Buffers described thus far may not be resized. The __DynamicBuffer__
+concept defines a storage type whose size can change. Beast uses
+dynamic buffers in interfaces where the amount of storage required
+to complete an operation is not known ahead of time.
+
+
+
+    [/ [heading Synchronous I/O] ]
```

Comment:
Is this heading supposed to be here?

Reply:
That's a commented-out heading, I am going to continue with these docs later